subsys/net/lib/websocket/websocket.c

```diff
@@ -443,12 +443,25 @@ static int websocket_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 }
 
 static void websocket_mask_payload(u8_t *payload, size_t payload_len,
-				   u32_t masking_value)
+				   u32_t masking_value, s8_t mask_shift)
 {
-	int i;
+	int *mask_shift_ptr;
+	int i, zero = 0;
+
+	/* Special handling when receiving data. As we cannot guarantee that
+	 * read data is received in four byte blocks, we need to pick proper
+	 * byte from the shift mask. When sending data, the mask_shift is set
```

Comment:
> When sending data, the mask_shift is set to -1

Why do we call this routine at all "when sending data"?

Reply:
Ah, nevermind, it's the structure of the operations here got me confused here. I just remember that masking of data happens in one direction, but not another (e.g. server->client, but not vice-versa, I don't remember which exactly). So, the question then is why we need this special -1 value, why don't just pass 0 as an arg.

Reply:
When unmasking data, we need to select correct byte from 32-bit mask value. This is an issue if we are reading less than 4 bytes. Anyway, I will refactor this this part in order to avoid these magic values.